src/main/java/com/demo/classroom/Validation/ValidPassword.java

```diff
@@ -0,0 +1,39 @@
+package com.demo.classroom.Validation;
+
+import jakarta.validation.Constraint;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import jakarta.validation.Payload;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Constraint(validatedBy = com.demo.classroom.Validation.ValidPassword.PasswordValidator.class)
+@Target({ ElementType.FIELD, ElementType.METHOD, ElementType.PARAMETER, ElementType.ANNOTATION_TYPE })
+@Retention(RetentionPolicy.RUNTIME)
+public @interface ValidPassword {
+
+    String message() default "Invalid password format";
+
+    Class<?>[] groups() default {};
+
+    Class<? extends Payload>[] payload() default {};
+
+    public static class PasswordValidator implements ConstraintValidator<ValidPassword, String> {
+        private static final String PASSWORD_PATTERN =
+            "^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%^&+=!])(?=\\S+$).{8,}$";
```

Comment:
Done

Reply:
did you test it properly?

Reply:
Yes i checked it. I notice that it is giving me the whole error message instead of saying what is missing like uppercase, lowercase or special character is required.  I refactored it. Its properly handled now.

Below is the postman screenshots: 
![passwrd-validation](https://github.com/user-attachments/assets/df56166d-77a4-4cf5-9751-c17ec2b27aa8)